### DISCUSSION TITLE
Why does HeavenMS derivatives need custom wz?

### DISCUSSION THREAD

**Discussion by weijiangan** [P0nk/Cosmic#235] [Q&A] [answered]
I kept finding that HeavenMS derivates require custom wz, and I noticed that in your README that the eventual goal is to be able to use vanilla ones. I've been googling and I can't find any good explanation. I assume you are aware of edits made? Can you shed some light on this?

**Comment by Arnuh** [comment, marked as answer]
HeavenMS itself needs a custom wz because it added custom changes, implemented features incorrectly, or in some cases fixed wz errors.

You can see why wz changes were added by just looking at the git history of the heavenms wz folder

**Reply by weijiangan** [reply]
So technically it's possible to put in vanilla wz and start fixing code in the server
